jax/_src/third_party/scipy/interpolate.py

```diff
@@ -0,0 +1,155 @@
+import scipy.interpolate as osp_interpolate
+
+from jax._src.tree_util import register_pytree_node
+from jax._src.numpy.lax_numpy import (_promote_dtypes_inexact, asarray,
+                                      broadcast_arrays, can_cast, empty, nan,
+                                      searchsorted, where, zeros)
+from jax._src.numpy.util import _wraps
+
+
+def _ndim_coords_from_arrays(points, ndim=None):
+  """Convert a tuple of coordinate arrays to a (..., ndim)-shaped array."""
+  if isinstance(points, tuple) and len(points) == 1:
+    # handle argument tuple
+    points = points[0]
+  if isinstance(points, tuple):
+    p = broadcast_arrays(*points)
+    for p_other in p[1:]:
+      if p_other.shape != p[0].shape:
+        raise ValueError("coordinate arrays do not have the same shape")
+    points = empty(p[0].shape + (len(points),), dtype=float)
+    for j, item in enumerate(p):
+      points = points.at[..., j].set(item)
+  else:
+    points = asarray(points)  # SciPy: asanyarray(points)
+    if points.ndim == 1:
+      if ndim is None:
+        points = points.reshape(-1, 1)
+      else:
+        points = points.reshape(-1, ndim)
+  return points
+
+
+@_wraps(
+    osp_interpolate.RegularGridInterpolator,
+    lax_description="""
+In the JAX version, `bounds_error` defaults to and must always be `False` since no
+bound error may be raised under JIT.
+
+Furthermore, in contrast to SciPy no input validation is performed.
+""")
+class RegularGridInterpolator:
+  # Based on SciPy's implementation which in turn is originally based on an
+  # implementation by Johannes Buchner
+
+  def __init__(self,
+               points,
+               values,
+               method="linear",
+               bounds_error=False,
+               fill_value=nan):
```

Comment:
Call `_check_arraylike()` on `values` here; from its use below it seems that it must be an array.